### PR TITLE
[#4031] Improve IGroupForm support

### LINKED ISSUE #4031: Improve IGroupForm support
Support for custom group schemas is well behind dataset ones.

These are some issues that need addressing:
* Issues around differentiating group-based and org-based plugins
* No `group_type` passed to templates, routes hardcoded to the core `group` and `organization` controllers
* Hardcoded logic in the group controller that prevents using custom names (eg `/publisher` or `/theme`)
* Can not define a fallback `IGroupForm` for organizations

### PR DESCRIPTION
**Pull request by amercader**
Fixes #4031

The individual commits are self-contained and pretty self-explanatory, but basically this fixes:

* Ability for org based plugins to be detected as such by the controller
* Ability to register a fallback org plugin
* Allow generation of URLs using custom types (eg `/publisher` or `/theme`). Right now `/group` and `/organization` were hardcoded everywhere.

Only thing missing is updating the bootstrap 2 templates.

I really think a version of this should be backported to 2.7. I think everything is pretty safe, even the changes in plugin logic, but let's discuss.

### CODE REVIEW

**Comment by amercader** [comment]
Tentatively flagging for backport, pending review

**Comment by wardi** [comment]
Looks great. Not sure about the backport though, what do you think about a short 2.8 release cycle instead?

File: ckan/templates-bs2/group/members.html [hunk 1]
```diff
@@ -25,10 +25,10 @@ <h3 class="page-heading">{{ _('{0} members'.format(c.members|length)) }}</h3>
         <td>{{ role }}</td>
         <td>
           <div class="btn-group pull-right">
-            <a class="btn btn-small" href="{% url_for controller='group', action='member_new', id=c.group_dict.id, user=user_id %}" title="{{ _('Edit') }}">
+            <a class="btn btn-small" href="{% url_for controller='group', action='member_delete', id=c.group_dict.id, user=user_id %}" title="{{ _('Edit') }}">
```
Comment (tino097) [inline ckan/templates-bs2/group/members.html]: @amercader action should me member_new

Comment (amercader) [reply ckan/templates-bs2/group/members.html]: Thanks @tino097, can you review and merge https://github.com/ckan/ckan/pull/4046? cheers

**Pull request merged**